File: 2022/20221213/Carpet.js
export function calsNeededYellow(brown, i) {
  return (brown / 2 - i - 2) * i;
}

export default function solution(brown, yellow) {
  const answer = [];

  for (let i = 1; i < brown; i += 1) {
    if (yellow === calsNeededYellow(brown, i)) {
      answer[0] = brown / 2 - i;
      answer[1] = i + 2;
      return answer;
    }
  }
  return answer;
}

File: 20230329/Carpet/src/main/java/Carpet.java
public class Carpet {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];

        for (int i = 1; i <brown; i += 1) {
            if (yellow == calcNecessaryYellow(brown, i)) {
                answer[0] = brown / 2 - i;
                answer[1] = i + 2;
                return answer;
            }
        }
        return answer;
    }

    public int calcNecessaryYellow(int brown, int i) {
        return (brown / 2 - i - 2) * i;
    }
}
